### ISSUE TITLE
1.14.4 BoP and Simple farming Support.

### ISSUE THREAD

**Issue by Wissi** [blay09/FarmingForBlockheads#63]
Awesome work, we love it so much,

Could you add **Biomes o' Plenty** and **Simple Farming** saplings, seeds and flowers to be loaded directly (Like the XL Food Mod)?

Thanks.

**Comment by Wissi** [comment]
Great mod we love your work.

We have custom datapacks for several mods, yet I just can't get yours to work through a datapack.

Using your "Market Configuration Example (for Minecraft 1.14)" and aren't able to get the other "inbuilt groups" to show up.
```
Current inbuilt group names (that can be used for group overrides) For Vanilla:
- Vanilla Flowers
- Vanilla Mushrooms
- BiomesOPlenty Saplings
```
Or even create a custom entry.
Tried in different "Folder Structures" (as per Minecraft's Datapacks wiki) and will load and recognize as a datapack in game, but your mod will not show any changes.
Folder Structures I tried (in world datapack folder):
```
\data\farmingforblockheads_compat\example.json
\data\farmingforblockheads_compat\recipes\example.json
\data\farmingforblockheads_compat\loot_tables\example.json
\data\farmingforblockheads_compat\tags\functions\example.json
```
Is there a specific _folder structure_ your mod is looking for the datapack or a _specific name_ for the .json file?

Full datapack file content is [here](https://gist.github.com/Wissi/e7d2af98557353f02eac542b426acf03).

Thank you so much for your great work, looking forward to add it to our modpacks.

**Comment by Nachtflame** [comment]
Also frustrated by the apparent lack of proper function, I did some testing, and seem to have figured it out - 

File structure _actually_ needs to be:
```
\data\farmingforblockheads\farmingforblockheads_compat\{files}.json
```

And the section for adding _customCategories_ completely invalidates all function.
I'd split the sections into individual files, and tried each in turn.  That section was the only one that resulted in an empty/bugged market.

Hope this helps!

**Comment by BlayTheNinth** [comment]
I had an error in the example json which was causing customCategories to fail loading + some missing error handling in the datapack loading. The latest version should be clearer about errors and I've also updated the [README](https://github.com/blay09/FarmingForBlockheads/blob/1.14.x/README.md#custom-market-entries) to include info on custom market data.

BiomesOPlenty support and SimpleFarming Support has been added (and can also act as an example for your own files, see https://github.com/blay09/FarmingForBlockheads/tree/1.14.x/src/main/resources/data/farmingforblockheads/farmingforblockheads_compat).
